workbench-core/infrastructure/src/workbenchCognito.ts

```diff
@@ -137,43 +132,47 @@ export class WorkbenchCognito extends Construct {
         callbackUrls: websiteUrls,
         logoutUrls: websiteUrls
       },
+      generateSecret: true,
       accessTokenValidity: props.accessTokenValidity,
       idTokenValidity: props.idTokenValidity,
-      refreshTokenValidity: props.refreshTokenValidity
+      refreshTokenValidity: props.refreshTokenValidity,
+      authFlows: {
+        userSrp: true,
+        userPassword: true,
+        custom: true
+      }
     };
-    const userPoolClientProps = merge(userPoolClientDefaults, tempUserPoolClientProps);
-    this.userPoolClient = new UserPoolClient(this, 'WorkbenchUserPoolClient', {
-      ...userPoolClientProps,
+    const webUiUserPoolClientProps = merge(userPoolClientDefaults, tempUserPoolClientProps);
+    this.webUiUserPoolClient = new UserPoolClient(this, 'WorkbenchUserPoolClient-webUi', {
+      ...webUiUserPoolClientProps,
       userPool: this.userPool,
-      userPoolClientName
-    });
-    this.userPool.identityProviders.forEach((provider) => this.userPoolClient.node.addDependency(provider));
-
-    const describeCognitoUserPoolClient = new AwsCustomResource(this, 'DescribeCognitoUserPoolClient', {
-      resourceType: 'Custom::DescribeCognitoUserPoolClient',
-      onCreate: {
-        region: Stack.of(this).region,
-        service: 'CognitoIdentityServiceProvider',
-        action: 'describeUserPoolClient',
-        parameters: {
-          UserPoolId: this.userPool.userPoolId,
-          ClientId: this.userPoolClient.userPoolClientId
-        },
-        physicalResourceId: PhysicalResourceId.of(this.userPoolClient.userPoolClientId)
-      },
-      policy: AwsCustomResourcePolicy.fromSdkCalls({
-        resources: [this.userPool.userPoolArn]
-      }),
-      installLatestAwsSdk: true
+      userPoolClientName: `${userPoolClientName}-webUi`
```

Comment:
We could have a "userPoolWebUiClientName" variable to make this more configurable.